### ISSUE TITLE
defercheck produces false positives with `switch arms`

### ISSUE THREAD

**Issue by tamird** [opennota/check#23]
running `defercheck` on [CockroachDB](https://github.com/cockroachdb/cockroach) results in:
```
$ defercheck ./...
./config: config/testutil.go:48:3: Repeating defer testingLock.Unlock() inside function TestingSetupZoneConfigHook
./kv: kv/txn_coord_sender.go:688:3: Repeating defer tc.cleanupTxn() inside function updateState
```

These are both false positives, but this issue is about the second one. Here's the relevant code: https://github.com/cockroachdb/cockroach/blob/e02b984b0d3b7d50e290151d87175a724c467852/kv/txn_coord_sender.go#L660:L688

```go
	case *roachpb.TransactionStatusError:
		// Likely already committed or more obscure errors such as epoch or
		// timestamp regressions; consider txn dead.
		defer tc.cleanupTxn(trace, t.Txn)
	case *roachpb.OpRequiresTxnError:
		panic("OpRequiresTxnError must not happen at this level")
	case *roachpb.ReadWithinUncertaintyIntervalError:
		// Mark the host as certain. See the protobuf comment for
		// Transaction.CertainNodes for details.
		if t.NodeID == 0 {
			panic("no replica set in header on uncertainty restart")
		}
		newTxn.Update(&t.Txn)
		newTxn.CertainNodes.Add(t.NodeID)
		// If the reader encountered a newer write within the uncertainty
		// interval, move the timestamp forward, just past that write or
		// up to MaxTimestamp, whichever comes first.
		candidateTS := newTxn.MaxTimestamp
		candidateTS.Backward(t.ExistingTimestamp.Add(0, 1))
		newTxn.Timestamp.Forward(candidateTS)
		newTxn.Restart(ba.GetUserPriority(), newTxn.Priority, newTxn.Timestamp)
		t.Txn = *newTxn
	case *roachpb.TransactionAbortedError:
		trace.SetError()
		newTxn.Update(&t.Txn)
		// Increase timestamp if applicable.
		newTxn.Timestamp.Forward(t.Txn.Timestamp)
		newTxn.Priority = t.Txn.Priority
		t.Txn = *newTxn
		// Clean up the freshly aborted transaction in defer(), avoiding a
		// race with the state update below.
		defer tc.cleanupTxn(trace, t.Txn)
```
